web/Services/LandlordService.cs

```diff
@@ -231,6 +239,35 @@ public async Task UnapproveLandlord(int landlordId)
         await _dbContext.SaveChangesAsync();
     }
 
+    public async Task<ILandlordService.DisableOrEnableLandlordResult> DisableOrEnableLandlord(int landlordId, string action)
+    {
+        var landlord = await GetLandlordIfExistsFromId(landlordId);
+        if (landlord is null)
+        {
+            return ILandlordService.DisableOrEnableLandlordResult.ErrorLandlordNotFound;
+        }
+
+        if (action == "disable")
+        {
+            if (landlord.Disabled)
+            {
+                return ILandlordService.DisableOrEnableLandlordResult.ErrorAlreadyInState;
+            }
+            landlord.Disabled = true;
+        }
+        else
+        {
+            if (!landlord.Disabled)
+            {
+                return ILandlordService.DisableOrEnableLandlordResult.ErrorAlreadyInState;
+            }
+            landlord.Disabled = false;
+        }

```

Comment:
2 things:
1. I think we shouldn't be accepting anything other than disable as enable
2. We can shorten this slightly

So the 2 together would be
```suggestion
        if ((action == "disable" && landlord.Disabled)
            || (action == "enable" && !landlord.Disabled))
        {
            return ILandlordService.DisableOrEnableLandlordResult.ErrorAlreadyInState;
        }

        landlord.Disabled = action switch
        {
            "disable" => true,
            "enable" => false,
            _ => landlord.Disabled
        };

```